### PR TITLE
[527522] Handle OperationCanceledException

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=527522

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java [hunk 1]
```diff
@@ -596,9 +597,14 @@ public Object getAdditionalProposalInfo(IProgressMonitor monitor) {
...
 				if (additionalProposalInfo instanceof Provider) {
-					Object o = ((Provider<?>) additionalProposalInfo).get();
-					if (o instanceof EObject)
-						eObject = (EObject) o;
+					try {
+						Object o = ((Provider<?>) additionalProposalInfo).get();
+						if (o instanceof EObject)
+							eObject = (EObject) o;
+					} catch (OperationCanceledException e) {
+						monitor.setCanceled(true);
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java]: Is this the common pattern?

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java]: i have doubts this is the correct solution

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java]: simply set o to null?

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java]: I second Christians concerns here. Usually the caller of a method that accepts an IProgressMonitor also describes the contract for the cancellation. Especially it should be able to deal with OCE.

**Comment by kthoms** [comment]
Thanks for your ideas, canceling this PR

**Pull request closed without merging**